cli/src/commands/config.rs

```diff
@@ -158,6 +176,12 @@ pub(crate) fn cmd_config_list(
             continue;
         }
 
+        if let Some(target_source) = args.get_source_kind() {
+            if target_source != *source {
+                continue;
+            }
+        }
+
```

Comment:
nit: perhaps, --user/--repo would conflict with --include-defaults. They all apply to the `source`.